### PR TITLE
Filter out noise messages from coda files

### PR DESCRIPTION
**Pull request by IsaackMwenda**
This is a temporary fix to filter out noise messages from coda files

### CODE REVIEW

File: filter_noise_messages.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    args = parser.parse_args()
+    coded_dir_path = args.coded_dir_path
+
+    with open(coded_dir_path) as f:
+        messages = json.load(f)
+
+    not_noise = []
+    def filter_noise_message(messages):
+        for message in messages:
+            if somali.DemographicCleaner.is_noise(message['Text'], min_length=10) == False:
```
Comment (as2388) [inline filter_noise_messages.py]:
```suggestion
            if not somali.DemographicCleaner.is_noise(message['Text'], min_length=10):
```

File: filter_noise_messages.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                        help="Directory to read then write coded Coda files to")
+
+    args = parser.parse_args()
+    coded_dir_path = args.coded_dir_path
+
+    with open(coded_dir_path) as f:
+        messages = json.load(f)
+
+    not_noise = []
+    def filter_noise_message(messages):
```
Comment (as2388) [inline filter_noise_messages.py]: Defining a function in the __name__ == "__main__" case with the same name as the file then calling it only once feels a bit redundant. I don't think you need to define a function, just outdent all the code below

File: filter_noise_messages.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+from core_data_modules.logging import Logger
+from core_data_modules.util import IOUtils
+
+log = Logger(__name__)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Filter out noise messages from Coda Files",
+                                     # Support \n and long lines
+                                     formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument("coded_dir_path", metavar="coded-dir-path",
```
Comment (as2388) [inline filter_noise_messages.py]: Doing it in place is a bit scary. Consider taking separate input and output directories

File: filter_noise_messages.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    def filter_noise_message(messages):
+        for message in messages:
+            if somali.DemographicCleaner.is_noise(message['Text'], min_length=10) == False:
+                not_noise.append(message)
+        log.info(f'Filtered {len(messages) - len(not_noise)} noise messages from {len(messages)} total messages')
+
+        # Output all RQA and Follow Up surveys messages which aren't noise to Coda
+        IOUtils.ensure_dirs_exist(coded_dir_path)
+        with open(coded_dir_path, 'w') as f:
+            json.dump(not_noise,f, indent=2)
```
Comment (as2388) [inline filter_noise_messages.py]:
```suggestion
            json.dump(not_noise, f, indent=2)
```

File: filter_noise_messages.py (hunk 1 above)
Comment (IsaackMwenda) [reply filter_noise_messages.py]: updated thanks

File: filter_noise_messages.py (hunk 2 above)
Comment (IsaackMwenda) [reply filter_noise_messages.py]: updated thanks

File: filter_noise_messages.py (hunk 3 above)
Comment (IsaackMwenda) [reply filter_noise_messages.py]: updated thanks

File: filter_noise_messages.py (hunk 4 above)
Comment (IsaackMwenda) [reply filter_noise_messages.py]: updated thanks

**Pull request merged**